### PR TITLE
refactor(snapshots): to their own crate

### PR DESCRIPTION
**Pull request by AaronO**
A revival of #10472, which is generally cleaner and facilitates cross-compiling deno, thus unblocking arm64 CI builds

### CODE REVIEW

**Review by piscisaureus — APPROVED** [review]

File: core/runtime.rs [hunk 1]
```diff
@@ -53,6 +53,16 @@ pub enum Snapshot {
   Boxed(Box<[u8]>),
 }
 
+impl Clone for Snapshot {
```
Comment (bartlomieju) [inline core/runtime.rs:56]: Why is it needed?

File: runtime/web_worker.rs [hunk 2]
```diff
@@ -442,7 +445,7 @@ impl WebWorker {
 
     let mut js_runtime = JsRuntime::new(RuntimeOptions {
       module_loader: Some(options.module_loader.clone()),
-      startup_snapshot: Some(js::deno_isolate_init()),
+      startup_snapshot: options.startup_snapshot.clone(),
```
Comment (bartlomieju) [inline runtime/web_worker.rs:448]: Use `.take()` instead of clone?

File: runtime/worker.rs [hunk 3]
```diff
@@ -154,14 +155,16 @@ impl MainWorker {
...
+      js::init(),
       // Permissions ext (worker specific state)
       perm_ext,
     ];
     extensions.extend(std::mem::take(&mut options.extensions));
 
     let mut js_runtime = JsRuntime::new(RuntimeOptions {
       module_loader: Some(options.module_loader.clone()),
-      startup_snapshot: Some(js::deno_isolate_init()),
+      startup_snapshot: options.startup_snapshot.clone(),
```
Comment (bartlomieju) [inline runtime/worker.rs:167]: Ditto

File: snapshots/build_runtime.rs [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+  let js_runtime = JsRuntime::new(RuntimeOptions {
+    will_snapshot: true,
+    extensions,
+    ..Default::default()
+  });
+  write_runtime_snapshot(js_runtime, snapshot_path);
+}
+
+// TODO(bartlomieju): this module contains a lot of duplicated
+// logic with `cli/build.rs`, factor out to `deno_core`.
```
Comment (bartlomieju) [inline snapshots/build_runtime.rs:63]: Seems like this comment is no longer valid?

**Review by bartlomieju — COMMENTED** [review]
@AaronO can you provide some rationale for this change? How does it impact build times and CI times?

**Review by bartlomieju — APPROVED** [review]
LGTM

**Review by littledivy — APPROVED** [review]

**Pull request merged**